providers/azure.go

```diff
@@ -192,6 +193,13 @@ func (p *AzureProvider) EnrichSession(ctx context.Context, s *sessions.SessionSt
 	}
 	s.Email = email
 
+	user, err := p.getUserFromProfileAPI(ctx, s.AccessToken)
+	if err != nil {
+		return fmt.Errorf("unable to get user: %v", err)
+	}
+	// Not going to throw an error if this is empty
+	s.User = user
```

Comment:
Currently, the user may not be empty, we should check before we overwrite it as I assume the ID Token value should take precedence if available (This is a convention within the proxy at least)

Also, you may need to adjust other parts of this function, looking at line 183, if the email is set but the user isn't, this part of the code is skipped, we probably don't want that right?

Reply:
Thanks a lot for pointing this out. I was under the impression that the session enrichment only happens in this file, but it looks like the object being carried over. I need to add an "or" clause in line 183 to make sure the function runs when email or name exists. as for rewriting, I can add the clause to not overwrite as well for sure.